### ISSUE TITLE
GPS: power off on EINT

### ISSUE THREAD

**Issue by gauteh** [gauteh/omb-2024-nano#1]
* https://github.com/sparkfun/SparkFun_u-blox_GNSS_Arduino_Library/blob/main/examples/Example22_PowerOff/Example22_PowerOff.ino
* https://learn.sparkfun.com/tutorials/gnss-receiver-breakout---max-m10s-qwiic-hookup-guide

**Comment by gauteh** [comment]
I'm seeing around 24 mA @ 5V when logging IMU. Seems a bit high to me? I guess I will see the low when the IMU is done sampling and it goes to sleep, would be helpful if the IMU sample window could be decreased to e.g. 30 seconds for in the test-mode.

**Comment by gauteh** [comment]
```
content: 0x47 | 0x1 | 0x46 | 0x5E | 0x16 | 0xB2 | 0x66 | 0xAB | 0xCF | 0xFD | 0x23 | 0x8E | 0x83 | 0x2D | 0x3 | 0x45 |
Rock7 format content: 4701465E16B266ABCFFD238E832D0345
--------------------
turn gnss off
putting gps to sleep.
Attempting to put gps to sleep
gnss wire started
success starting GNSS
gps sleeping...
gps sleeping.
success.
turn iridium on
turn gnss off
putting gps to sleep.
Attempting to put gps to sleep
gnss wire started
success starting GNSS
gps sleeping...
gps sleeping.
success.
Calling internalBegin
IridiumSBD::beginSerialPort
>> AT
Waiting for response OK

```

**Comment by gauteh** [comment]
Yes, that is what I'm trying to get them to do :)

**Comment by gauteh** [comment]
Pretty sure it's the GPS, because unplugging the LiIon on the rockblock does not change the current draw.

**Comment by gauteh** [comment]
Relevant code is here: https://github.com/gauteh/omb-2024-nano/blob/main/src/gnss_manager.cpp#L7

**Comment by gauteh** [comment]
> the change is to hard code taking just a few imu measurements:
> 
> https://github.com/jerabaul29/OpenMetBuoy-v2021a/blob/129916e1e05f97facc37a9b5c94781a2f2e4fec0/legacy_firmware/firmware/functionality_test_mode/wave_analyzer.cpp#L62

Thanks, I have to do this, since lots of ports and pins are changed from original OMB.

**Comment by gauteh** [comment]
Can't unplug the GNSS, its mounted on board.

**Comment by gauteh** [comment]
Are the measurements above with the power led on?

**Comment by gauteh** [comment]
I think that will destroy the board, I should probably just try this: https://github.com/sparkfun/SparkFun_u-blox_GNSS_Arduino_Library/blob/main/examples/Example22_PowerOff/Example22_PowerOff.ino

**Comment by gauteh** [comment]
Yes, good point. I can remove the Artemis to test that. There are also some broken artemises, that use much more power.

**Comment by gauteh** [comment]
https://github.com/sparkfun/SparkFun_u-blox_GNSS_Arduino_Library/issues/228

**Comment by gauteh** [comment]
If this dosen't work we need to add some on/off circuit.

**Comment by gauteh** [comment]
Maybe @zerblatt007 has the chance to help us set up the MOSFET circuit, here's the schematic from SparkFun: https://cdn.sparkfun.com/assets/f/4/5/3/4/SparkFun_Artemis_Global_Tracker_SCHEMATIC_v10.pdf ?

**Comment by zerblatt007** [comment]
GPS power switch should be easy enough.. :)

**Comment by zerblatt007** [comment]
I have inserted the Q1 FET on A16 and updated the PCB, just to see how
things pan out.
This is just more or less dumped on there, so check if it looks ok to you..
There was a new R5 also? And an extra BAT2 ..





fre. 30. aug. 2024 kl. 08:43 skrev Roger Storvik ***@***.***>:

> GPS power switch should be easy enough.. :)
>

**Comment by gauteh** [comment]
Nice. Adding the artemis module directly (with USB to serial) would be next on the wishlist ! (should have a JTAG header holes just in case as well).
